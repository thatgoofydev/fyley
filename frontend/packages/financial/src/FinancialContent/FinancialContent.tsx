import React, { FunctionComponent } from "react";
import { Switch } from "react-router";
import { Route, useRouteMatch } from "react-router-dom";
import { Page } from "@fyley/ui-lib";
import { AccountPage } from "@fyley/financial-accounts";

export const FinancialContent: FunctionComponent = () => {
  const { path } = useRouteMatch();

  return (
    <Switch>
      <Route path={`${path}/overview`}>
        <Page title="Overview">financial overview page</Page>
      </Route>
      <Route path={`${path}/accounts`}>
        <AccountPage />
      </Route>
    </Switch>
  );
};
