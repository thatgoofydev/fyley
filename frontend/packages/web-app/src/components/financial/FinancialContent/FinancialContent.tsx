import React, { FunctionComponent } from "react";
import { Route, Switch } from "react-router-dom";
import { AccountsPage } from "../Accounts/AccountsPage/AccountsPage";

export const FinancialContent: FunctionComponent = () => {
  return (
    <Switch>
      <Route path="/financial/overview">overview</Route>
      <Route path="/financial/accounts">
        <AccountsPage />
      </Route>
    </Switch>
  );
};
