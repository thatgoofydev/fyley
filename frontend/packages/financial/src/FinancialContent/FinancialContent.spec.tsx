import React from "react";
import { render, cleanup } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";
import { FinancialContent } from "./FinancialContent";

afterEach(cleanup);

describe("FinancialContent", function () {
  it("should render", function () {
    const { container } = render(
      <MemoryRouter initialEntries={["/financial/overview"]}>
        <Route path="/financial">
          <FinancialContent />
        </Route>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
