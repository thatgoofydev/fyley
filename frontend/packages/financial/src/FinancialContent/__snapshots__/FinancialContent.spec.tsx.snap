// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialContent should render 1`] = `
<div>
  <div
    class="Page-module_pageContainer__26bqU"
  >
    <header
      class="Page-module_pageHeader__36DgL"
    >
      <h1>
        Overview
      </h1>
    </header>
    <div
      class="Page-module_pageContent__3_kgD"
    >
      financial overview page
    </div>
  </div>
</div>
`;
